src/Rector/Property/DoctrineTargetEntityStringToClassConstantRector.php

```diff
@@ -112,6 +112,13 @@ private function changeTypeInAttributeTypes(Property $property, bool $hasChanged
             return $hasChanged ? $property : null;
         }
 
+        $property = $this->changeTypeInAttribute($attribute, $property);
+
+        return $hasChanged ? $property : null;
```

Comment:
better to check (! $hasChanged) early so no need `changeTypeInAttribute()` when not needed:

```php
if (! $hasChanged) {
    return null;
}

return $this->changeTypeInAttribute($attribute, $property);
```

Reply:
Interestingly this does not lead to the same result. I guess that `$property` is passed as reference and mutated in that process, even if `$hasChanged` is false..

Reply:
After staring at the code for several minutes, I think I finally figured it out. The `$hasChanged` Flag in `changeTypeInAttributeTypes()` is an indicator if this was already mutated by an Annotation Matcher. If it was detected as Annotation, there is no need to analyze it as Attribute as well.
So the condition should be:
```php
if ($hasChanged) {
    return null;
}

return $this->changeTypInAttribute($attribute, $property);
```

Reply:
I made it even clearer now, because `changeTypInAttribute()` doesn't need to know if the property was resolved as Annotation before.. See https://github.com/rectorphp/rector-doctrine/pull/114/commits/82653899217e6be7c7977f2c73c5b7aa4591d87d